net/portmapper/portmapper.go

```diff
@@ -749,9 +749,16 @@ func (c *Client) Probe(ctx context.Context) (res ProbeResult, err error) {
 		// See https://github.com/tailscale/tailscale/issues/3197 for
 		// an example of a device that strictly implements UPnP, and
 		// only responds to multicast queries.
+		//
+		// Then we send a discovery packet looking for
+		// urn:schemas-upnp-org:device:InternetGatewayDevice:1 specifically, not
+		// just ssdp:all, because there appear to be devices which only send
+		// their first descriptor (like urn:schemas-wifialliance-org:device:WFADevice:1)
+		// in response to ssdp:all. https://github.com/tailscale/tailscale/issues/3557
 		metricUPnPSent.Add(1)
 		uc.WriteTo(uPnPPacket, upnpAddr)
 		uc.WriteTo(uPnPPacket, upnpMulticastAddr)
+		uc.WriteTo(uPnPIGDPacket, upnpMulticastAddr)
```

Comment:
Bit of a bummer that we're up to 5 packets now (PMP + PCP + 3 UPnP?) for portmapping discovery. We'll probably want a better scheduler for these at some point that's more aware of what's worked/not in the past. I forget how smart our current one is, but I suspect not great.

Oh well.

Reply:
I think we can send SSDP way less often than we currently do, whether portmap protocols respond shouldn't change often. Maybe probe for all protocols a couple times shortly after startup.